src/frontend/src/handler/alter_relation_rename.rs

```diff
@@ -197,6 +200,105 @@ pub async fn handle_rename_source(
     Ok(PgResponse::empty_result(StatementType::ALTER_SOURCE))
 }
 
+pub async fn handle_rename_schema(
+    handler_args: HandlerArgs,
+    schema_name: ObjectName,
+    new_schema_name: ObjectName,
+) -> Result<RwPgResponse> {
+    let session = handler_args.session;
+    let db_name = session.database();
+    let schema_name = Binder::resolve_schema_name(schema_name)?;
+    let new_schema_name = Binder::resolve_schema_name(new_schema_name)?;
+
+    let schema_id = {
+        let user_reader = session.env().user_info_reader().read_guard();
+        let catalog_reader = session.env().catalog_reader().read_guard();
+        let schema = catalog_reader.get_schema_by_name(db_name, &schema_name)?;
+        let db_id = catalog_reader.get_database_by_name(db_name)?.id();
+
+        // The current one should not be system schema.
+        if is_system_schema(&schema.name()) {
```

Comment:
check new name as well?

Reply:
If new name is the same as system schema, it will be ruled out by duplication check.